eskip/eskip.go

```diff
@@ -591,11 +592,28 @@ func newRouteDefinition(r *parsedRoute) (*Route, error) {
 	return rd, err
 }
 
+type eskipLexParser struct {
+	lexer  eskipLex
+	parser eskipParserImpl
+}
```

Comment:
Parser consists of 1+16 `eskipSymType` structs
https://github.com/zalando/skipper/blob/77f4c5e868917a90a8823d734e7186fb7a34140d/eskip/parser.go#L200-L204
which is quite large
https://github.com/zalando/skipper/blob/77f4c5e868917a90a8823d734e7186fb7a34140d/eskip/parser.go#L21-L44

A new lexer is always used with the parser so it makes sense to pool them together.

`eskipSymType` is in fact a union type, i.e. only one of the fields is used on each parse step.
I've experimented with implementing it as a 
```yacc
%union {
	val interface{}
}
```
and a set of converters like
```go
func (u *eskipSymType) token() string        { return u.val.(string) }
func (u *eskipSymType) stringvals() []string { return u.val.([]string) }
func (u *eskipSymType) route() *parsedRoute { return u.val.(*parsedRoute) }
...
```
similar to how it is done in [cockroachdb](https://github.com/cockroachdb/cockroach/blob/c21c90f93219b857858518d25a8bc061444d573c/pkg/sql/parser/sql.y#L205-L249).

This brings some improvement but requires quite heavy rewrite of parser.y grammar rules.

I have also couple of lexer refactorings that eliminate inefficiencies and reduce allocations but not as much as pooling lexer and parser. These could be done in subsequent PRs.